web-client/src/views/Messages/MessageTable.tsx

```diff
@@ -0,0 +1,401 @@
+        messagesIndividualInboxHelper.someMessagesSelected &&
+        !messagesIndividualInboxHelper.allMessagesSelected;
+    }, [
+      selectAllCheckboxRef.current,
+      messagesIndividualInboxHelper.someMessagesSelected,
+      messagesIndividualInboxHelper.allMessagesSelected,
+    ]);
+
+    // For cases when messages can be selected in the given view
+    const getSelectAllCheckbox = () => {
+      return (
+        <th>
+          <input
+            aria-label="all-messages-checkbox"
+            checked={messagesIndividualInboxHelper.allMessagesSelected}
+            data-testid="all-messages-checkbox"
+            disabled={!messagesIndividualInboxHelper.allMessagesCheckboxEnabled}
+            id="all-messages-checkbox"
+            ref={selectAllCheckboxRef}
+            type="checkbox"
+            onChange={() => {
+              const selectAll = formattedMessages.messages.map(message => ({
+                messageId: message.messageId,
+                parentMessageId: message.parentMessageId,
+              }));
+              setSelectedMessagesSequence({ messages: selectAll });
+            }}
+          />
+        </th>
+      );
+    };
+
+    // For cases when messages can be completed in the given view
+    const getCompleteAllButton = () => {
+      return (
+        <div className="desktop:grid-col-4 tablet:grid-col-12 tablet:margin-top-2 text-right">
+          <Button
+            link
+            className="action-button"
+            data-testid="message-batch-mark-as-complete"
+            disabled={!messagesIndividualInboxHelper.isCompletionButtonEnabled}
+            icon="check-circle"
+            id="button-batch-complete"
+            onClick={() => {
+              batchCompleteMessageSequence();
+            }}
+          >
+            Complete
+          </Button>
+        </div>
+      );
+    };
+
+    return (
+      <>
+        <SuccessNotification />
+        <ErrorNotification />
+        {screenMetadata.completionSuccess && (
+          <div
+            aria-live="polite"
+            className="usa-alert usa-alert--success"
+            data-testid="message-detail-success-alert"
+            role="alert"
+          >
+            <div className="usa-alert__body">
+              Message(s) completed at{' '}
+              {messagesIndividualInboxHelper.messagesCompletedAt} by{' '}
+              {messagesIndividualInboxHelper.messagesCompletedBy}
+            </div>
+          </div>
+        )}
+        <div className="overflow-x-auto" id={id}>
+          <div className="grid-row grid-gap">
+            {messageFilters.length > 0 && (
+              <div className="desktop:grid-col-8 tablet:grid-col-12 display-flex flex-align-center">
+                <TableFilters
+                  filters={messageFilters}
+                  onSelect={updateMessageFilterSequence}
+                ></TableFilters>
+              </div>
+            )}
+            {selectable && getCompleteAllButton()}
```

Comment:
Coupling selectable and completable ... but for now, I think that's ok. We can separate them if needed.